### PR TITLE
Support Selenium v3.14

### PR DESCRIPTION
**Pull request by TroyWalshProf**
## Change list
Move to .Net Standard
Support Selenium 3.14 
 
## Types of changes

What types of changes are you proposing/introducing to .NET client?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Details
Make project .Net Standard.
- The will require those using the .Net Framework to update to at least 4.6.1 
Add AppiumOptions
- Mirror what Selenium's move away from desired capabilities
- See https://github.com/SeleniumHQ/selenium/issues/6278#issuecomment-420728084 for more info
Add AppiumCapabilities 
- Create Appium specific capability class

### CODE REVIEW

File: appium-dotnet-driver/Appium/AppiumDriver.cs [hunk 1]
```diff
@@ -40,50 +40,50 @@ public abstract class AppiumDriver<W> : RemoteWebDriver, IExecuteMethod, IFindsB
...
+        public AppiumDriver(ICommandExecutor commandExecutor, ICapabilities AppiumOptions)
+            : base(commandExecutor, AppiumOptions)
         {
             AppiumCommand.Merge(commandExecutor.CommandInfoRepository);
             ElementFactory = CreateElementFactory();
         }
 
-        public AppiumDriver(ICapabilities desiredCapabilities)
-            : this(AppiumLocalService.BuildDefaultService(), desiredCapabilities)
+        public AppiumDriver(ICapabilities AppiumOptions)
```
Comment (mykola-mokhnach) [inline appium-dotnet-driver/Appium/AppiumDriver.cs]: should `appiumOptions` start with a capital letter everywhere?

**Comment by TroyWalshProf** [comment]
I actually did think about that.  
From a maintenance perspective I think we need an AppiumOptions class. 
*I am confident that this code will need to be updated when DesiredCaps are no longer available.
That being said, if this change goes through I could see us extending it with iOS, Android, Windows and Tizen specific option classes. That way we get the best of both world.

**Comment by akinsolb** [comment]
Sure. This takes a task out of my weekend. Descendants can always be done later if necessary., 
Did you manage to run a test against this branch? Atm, the CI doesn't have verification tests yet but running the last written test should sufficient, I think that would be the IOSLockDeviceTest in order to merge.

**Comment by TroyWalshProf** [comment]
@jlipps - This is defiantly a breaking change.
1. Changes AppiumDriver (and iOS, Android, etc.) constructor signatures
2. Removes support for .Net 4.6 and below, but adds support for .Net Core, Mono, Xamarin and UWP

Moving over to use the DriverOptions is based off an [interaction ](https://github.com/SeleniumHQ/selenium/issues/6278#issuecomment-420728084) with Jim Evens.
*Seems like the best long term strategy for remaining compatible with future versions of Selenium.

**Comment by jlipps** [comment]
@TroyWalshProf @akinsolb in that case, full steam ahead! Selenium and @jimevans compatibility are very important.

**Comment by jimevans** [comment]
TL;DR I’m in favor of this PR, but some thought should be put into the design of this part of the Appium .NET API going forward. 

I’m an outsider who is offering an  unsolicited opinion on a project whose community I’m not an active member of. Take the below with a grain of salt.

Dropping support for .NET Framework and moving to .NET Core only at this time would be a mistake, in my mostly irrelevant opinion. As mentioned by @akinsolb, the versions of `PageFactory` that are currently available (formerly in `WebDriver.Support`, now deprecated and migrated to the DotNetSeleniumExtras project) are not available in .NET Standard/Core projects. 

Moreover, and this is simply a philosophical difference, this change appears to be simply substituting one “allow arbitrary names and values with no validity checking” property bag for another. The whole point of moving from `DesiredCapabilities` to a type-safe options class in the first place was to provide guidance and prevent user frustration because the capability being requested used the wrong (arbitrary string) name or the wrong (data type or invalid) value. Again, if the Appium project doesn’t care about that user experience, and it’s a perfectly legitimate choice not to, that’s a fine approach. However, I would put some thought into the API before making this the last word on the subject.

To forestall the inevitable question about why such a change is needed in the first place, correct use of DesiredCapabilities has been a source of confusion for users in all but the simplest of cases. It results in cargo cult, copy-paste code with no understanding of why the arbitrary capability is being set or if it’s even appropriate and supported by the target driver. Coupled with the W3C WebDriver Specification’s strictness around capabilities in the protocol, and the ability of implementations to reject connection requests with unknown or invalid capabilities, making the users’ and maintainers’ lives a little easier by preventing those types of errors in the first place seems like an obvious answer. 

Finally, and this is simply by way of communication with a large downstream project about future possibilities, I’m thinking about ways to make the browser-specific `DriverOptions` API clearer, and `AddAdditionalCapability` is likely to be changing, which probably means deprecation, to be replaced by a new method or methods. I need a better way to solve the problem where a user wants to add a capability which belongs to a browser-specific driver dictionary (e.g., “goog:chromeOptions”, “moz:firefoxOptions”, etc.) and a global or top-level capability (like SauceLabs or BrowserStack user and API key capabilities). Right now, specific options classes have an overload for the method, but this is neither discoverable nor elegant, and the size of the user base using the imprecise code makes it impossible to just change the behavior to something that feels more correct.

**Comment by TroyWalshProf** [comment]
@akinsolb - Added .Net 4.5 support back.

**Comment by TroyWalshProf** [comment]
@akinsolb - This change also addresses 
https://github.com/appium/appium-dotnet-driver/issues/227
https://github.com/appium/appium-dotnet-driver/issues/191
*.Net Core can use .Net Standard libraries

**Comment by theAshenfire** [comment]
Hi,
Is there an ETA when this change will be available via NuGet?

**Comment by akinsolb** [comment]
@theAshenfire This is [available](
https://www.nuget.org/packages/Appium.WebDriver/4.0.0.3-beta) in the beta version

**Comment by jshinn6788** [comment]
What exactly is the resolution to the "Desired Capabilities" deprecated error?  because I'm still getting this error.

**Comment by TroyWalshProf** [comment]
@jshinn6788 - Desired capabilities for Selenium has been deprecated.
What they did in the Selenium project was to  create driver specific options.
See: https://github.com/SeleniumHQ/selenium/blob/16559ddb469ec45bbdfef062a5638fb70cb18224/java/client/test/org/openqa/selenium/chrome/ChromeOptionsFunctionalTest.java#L89

This  pattern was followed in Appium.
Now instead of using Desired capabilities you use AppiumOptions.
See: https://github.com/appium/appium-dotnet-driver/blob/1dcb236576a92990c2580b0e600e6c9f3a841c23/test/integration/Windows/StickyNotesTest.cs#L53

**Comment by jshinn6788** [comment]
Awesome!
This was very helpful
Thank you
